### PR TITLE
Trustonic pkcs11 fix build

### PR DESCRIPTION
**Pull request by AlexandreGonzalo**
Hi OpenSC team,
In my build environment, I don't always have pthread, thus sometimes, I don't have HAVE_PTHREAD defined.
I had a build issue.
Regards,
Alexandre.

### CODE REVIEW

File: src/tools/pkcs11-tool.c [hunk 1]
```diff
@@ -263,8 +263,8 @@ static const struct option options[] = {
 #ifndef _WIN32
 	{ "test-fork",		0, NULL,		OPT_TEST_FORK },
 #endif
-	{ "use-locking",	0, NULL,		OPT_USE_LOCKING },
 #if defined(_WIN32) || defined(HAVE_PTHREAD)
+	{ "use-locking",	0, NULL,		OPT_USE_LOCKING },
```
Comment (Jakuje) [inline src/tools/pkcs11-tool.c:267]: I think this should stay as it is, but the check should come inside of the handling of this operation to provide better error than "unknown option".

**Review by Jakuje — APPROVED** [review]

File: src/tools/pkcs11-tool.c (hunk 1 above)
Comment (AlexandreGonzalo) [reply src/tools/pkcs11-tool.c:267]: The issue is that OPT_USE_LOCKING is not defined when HAVE_PTHREAD is not defined.

Comment (Jakuje) [reply src/tools/pkcs11-tool.c:267]: Right. It would need some more modifications. So lets keep it this way.

**Comment by Jakuje** [comment]
can you rebase this on top of the current master so I can cleanly merge the outstanding patch?

**Comment by AlexandreGonzalo** [comment]
> can you rebase this on top of the current master so I can cleanly merge the outstanding patch?

Sure, it is done.

**Pull request merged**